eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelFeatures.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2021 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.channel
+
+import fr.acinq.eclair.Features.{AnchorOutputs, StaticRemoteKey, Wumbo}
+import fr.acinq.eclair.transactions.Transactions.{AnchorOutputsCommitmentFormat, CommitmentFormat, DefaultCommitmentFormat}
+import fr.acinq.eclair.{Feature, FeatureSupport, Features}
+
+/**
+ * Created by t-bast on 24/06/2021.
+ */
+
+/**
+ * Subset of Bolt 9 features used to configure a channel and applicable over the lifetime of that channel.
+ * Even if one of these features is later disabled at the connection level, it will still apply to the channel until the
+ * channel is upgraded or closed.
+ */
+case class ChannelFeatures(features: Features) {
+
```

Comment:
Should we add a `require` that all features are mandatory?

Reply:
We can, but that's not enforced when you `copy`, so it would only be a partial protection

Reply:
How about changing the signature to:

```scala
case class ChannelFeatures(activated: Set[Feature], unknown: Set[UnknownFeature])
```

It removes all the ambiguity and makes the choice between optional/mandatory something to only deal with at the codec level.